File: js/platform/src/modules/agent.ts
import { http } from "../core/http.js";
import { Endpoints } from "../core/endpoints.js";
import type { ApiResult } from "../core/result.js";

export const agent = {
  validate(definition: unknown): Promise<ApiResult<Record<string, unknown>>> {
    return http.postMap(Endpoints.agentValidate, definition);
  },

  getTemplates(): Promise<ApiResult<Record<string, unknown>>> {
    return http.getMap(Endpoints.agentTemplates);
  },

  execute(body: {
    userId: string;
    workflowRunId: string;
    definition: unknown;
    triggerInput: unknown;
    resumeStepKey?: string;
    resumeValue?: unknown;
    resumeContext?: Record<string, unknown>;
    executionTarget?: "handoff" | "platform";
  }): Promise<ApiResult<Record<string, unknown>>> {
    return http.postMap(Endpoints.agentExecute, body);
  },

  listRuns(): Promise<ApiResult<unknown[]>> {
    return http.getList(Endpoints.agentRuns);
  },

  getRun(id: string): Promise<ApiResult<Record<string, unknown>>> {
    return http.getMap(Endpoints.agentRunDetail(id));
  },
};

File: flutter/platform/lib/src/modules/agent.dart
import '../core/http_client.dart';
import '../core/api_endpoints.dart';
import '../core/result.dart';

/// Agent module — workflow validation, templates, execution, runs.
class AgentModule {
  const AgentModule();

  Future<ApiResult<Map<String, dynamic>>> validate(Map<String, dynamic> definition) =>
      HttpClient.instance.postMap(ApiEndpoints.agentValidate, body: definition);

  Future<ApiResult<Map<String, dynamic>>> getTemplates() =>
      HttpClient.instance.getMap(ApiEndpoints.agentTemplates);

  Future<ApiResult<Map<String, dynamic>>> execute(Map<String, dynamic> body) =>
      HttpClient.instance.postMap(ApiEndpoints.agentExecute, body: body);

  Future<ApiResult<List<dynamic>>> listRuns() =>
      HttpClient.instance.getRawList(ApiEndpoints.agentRuns);

  Future<ApiResult<Map<String, dynamic>>> getRun(String id) =>
      HttpClient.instance.getMap(ApiEndpoints.agentRunDetail(id));
}
